### PR TITLE
Method conST

### PR DESCRIPTION
**Pull request by heylf**
This tackles the issue #20 to add conST as a method.

I contacted the authors and they have added MIT license. 

Please check the two TODOs. The method has generally standard processing. GPU is not necessarily required but recommended.

### CODE REVIEW

File: method/conST/conST.py [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+device = 'cuda:0' if torch.cuda.is_available() else 'cpu'
+print('Using device: ' + device)
+params.device = device
+
+os.environ['PYTHONHASHSEED'] = str(seed)
+torch.backends.cudnn.benchmark = False
+torch.backends.cudnn.deterministic = True
+
+# TODO this is what conST uses for data preprocessing
+def adata_preprocess(i_adata, min_cells=3, pca_n_comps=300):
```
Comment (niklasmueboe) [inline method/conST/conST.py:176]: If they run this with 300 PCs we should probably keep this separate from the preprocessing, I don' think we will scale the numbers of PCs that high. Also I am sceptical about how much information these PCs are adding.

File: method/conST/config/config_1.json
Comment (niklasmueboe) [inline method/conST/config/config_1.json:1]: Can we reduce the number of parameters to the ones we are likely to actually change? I think for most of the other methods we stuck to keeping the architectural parameters at theri defaults.

File: method/conST/config/config_1.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    "cont_l2c": 0.1,
+    "cont_l2g": 0.1,
+    "edge_drop_p1": 0.1,
+    "edge_drop_p2": 0.1,
+    "node_drop_p1": 0.2,
+    "node_drop_p2": 0.3,
+    "eval_resolution": 1,
+    "eval_graph_n": 20,
+    "refinement": false,
+    "shape": "hexagon"
```
Comment (niklasmueboe) [inline method/conST/config/config_1.json:34]: The shape should probably be detected from the technology.

**Review by niklasmueboe — CHANGES_REQUESTED** [review]

**Comment by heylf** [comment]
@niklasmueboe finished your review

**Review by niklasmueboe — APPROVED** [review]

**Pull request merged**